evalml/tests/component_tests/decomposer_tests/test_stl_decomposer.py

```diff
@@ -38,39 +38,57 @@ def test_stl_raises_warning_high_smoother(caplog, ts_data):
     assert "STLDecomposer may perform poorly" in caplog.text
 
 
+@pytest.mark.parametrize(
+    "period,freq",
+    [
+        (7, "D"),
+        (30, "D"),
+        (365, "D"),
+        (12, "M"),
+        (40, "M"),
+    ],
+)
+def test_stl_sets_determined_period(
+    period,
+    freq,
+    generate_seasonal_data,
+):
+
+    X, y = generate_seasonal_data(real_or_synthetic="synthetic")(
+        period,
+        freq_str=freq,
+    )
+
+    stl = STLDecomposer()
+    stl.fit(X, y)
+    assert period * 0.99 <= stl.period <= period * 1.01
```

Comment:
can you add a comment why we use `0.99` and `1.01`? Looks somewhat arbitrary unless you have the context!

Reply:
I'm assuming there's a margin of error with the period determination that this test accounts for?